File: Python_code_challenges/fourLetterWords.py
""" fourLetterWords
Write a function that takes a sentence and returns the number of four letter
words it contains. Dont worry about handling punctuation 
"""

# If punctuation doesn't matter (as problem states):
def four_letter_words(s):
    count = 0
    arr = s.split()
    for w in arr:
        if len(w) == 4:
            count += 1
    return count


# Alternative: if punctuation matters and have to be accounted for:
import re
def four_letter_words(s):
    count = 0
    arr = re.split(r'\W+', s)
    for w in arr:
        if len(w) == 4:
            count += 1
    return count
    
    
#alternative, if punctuation matters (slower but more specific):
def four_letter_words(s):
    res, temp = 0, 0
    for w in s.split():
        for c in w:
            if c.isalpha(): 
                temp += 1
        if temp == 4: 
            res += 1
        temp = 0
    return res


print(four_letter_words("This sentence is fine")) # 2
print(four_letter_words("So is this one")) # 1
print(four_letter_words("Hello")) # 0
print(four_letter_words("This sentence is different, it has punctuation!!!")) # 1
print(four_letter_words("This one is tricky !!!!")) # 1 #this test is for variants where punctuation matters

File: PHP/PHP_challenges/fourLetterWords.php
<?php

/* fourLetterWords
Write a function that takes a sentence and returns the number of four letter
words it contains. Dont worry about handling punctuation 
*/


# If punctuation doesn't matter (as problem states):
function four_letter_words($s) {
    $res = 0;
    $words = preg_split('/\s+/', $s);
    foreach($words as $w) {
        if (strlen($w) === 4) 
            $res += 1;
    }
    return $res;
}


/* Alternative: if punctuation matters and have to be accounted for:

function four_letter_words($s) {
    $res = 0;
    $words = preg_split('/\W+/', $s);
    foreach($words as $w) {
        if (strlen($w) === 4) 
            $res += 1;
    }
    return $res;
}

*/


echo four_letter_words("This sentence is fine"), "\n"; # 2
echo four_letter_words("So is this one"), "\n"; # 1
echo four_letter_words("Hello"), "\n"; # 0
echo four_letter_words("This sentence is different, it has punctuation!!!"), "\n"; # 1
#echo four_letter_words("This one is tricky !!!!"), "\n"; # 1 #this test is for variants where punctuation matters
